xm-commons-topic/src/main/java/com/icthh/xm/commons/topic/config/ConsumerRecoveryCallback.java

```diff
@@ -64,6 +83,17 @@ public Object recover(RetryContext context) {
         return null;
     }
 
+    private static Headers addExceptionHeaders(RetryContext context, ConsumerRecord<?, ?> record) {
+        Headers headers = new RecordHeaders(record.headers());
+        Throwable e = context.getLastThrowable();
+        if (e instanceof ListenerExecutionFailedException && e.getCause() != null) {
+            e = e.getCause();
+        }
+        headers.add(new RecordHeader(EXCEPTION_MESSAGE, e.toString().getBytes(UTF_8)));
+        headers.add(new RecordHeader(EXCEPTION_STACKTRACE, getStackTrace(e).getBytes(UTF_8)));
```

Comment:
why bytest? it's more optimal way?

Reply:
because headers in kafka api it`s map<string, byte[]>